### PR TITLE
data-menu: enforce layer order

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request implements a `sort_by` option on the `LayerSelect` component, which defaults to the previous behavior of `icon` (order layer was _added_), but also supports `zorder` (shows layers on TOP first).  The `zorder` ordering is then used for the legend, data-menu, ~and plot options tabs~.

~Note that for plot options, it is possible to subscribe to layers from multiple viewers, and they may have different zorder.  In this case, the zorders are parsed and sorted based on the first viewer they are found in according to the selected viewers.~

NOTE: this now defers re-ordering plot options tab since we need to give more thought on how the RGB presets logic should work in this case.

https://github.com/user-attachments/assets/cfd7069a-cfd2-4782-a06d-68409f4f0cdd

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/template_mixin.py [hunk 1]
```diff
@@ -1439,6 +1441,8 @@ def __init__(self, plugin, items, selected, viewer,
         default_mode : str, optional
             What mode to use when making the default selection.  Valid options: first, default_text,
             empty.
+        sort_by : str, optional
+            How to sort the ordering of items.  Valid options: zorder, icon
```
Comment (rosteen) [inline jdaviz/core/template_mixin.py:1445]: Could you explain what the `icon` sorting behavior is here? Zorder is probably pretty self-explanatory but it's not immediately clear to me what the other option is.

Comment (kecnry) [reply jdaviz/core/template_mixin.py:1445]: I'll add it in the docstring as well - but technically its alphabetical ordering of the icons, which effectively means order in which the layers were first created and assigned their icons.

**Review by rosteen — APPROVED** [review]
LGTM now.

**Review by pllim — APPROVED** [review]
Code looks like code. But do we need PO approval given different people have different preferences of ordering?

**Comment by kecnry** [comment]
This was discussed and planned in advance

**Pull request merged**